lib/einhorn/command.rb

```diff
@@ -589,17 +589,30 @@ def self.replenish_gradually(max_unacked=nil)
       if seconds_ago > spinup_interval
         unacked = Einhorn::WorkerPool.unacked_unsignaled_modern_workers.length
         if unacked >= max_unacked
-          Einhorn.log_debug("There are #{unacked} unacked new workers, and max_unacked is #{max_unacked}, so not spinning up a new process")
+          Einhorn.log_info("There are #{unacked} unacked new workers, and max_unacked is #{max_unacked}, so not spinning up a new process")
         else
-          msg = "Last spinup was #{seconds_ago}s ago, and spinup_interval is #{spinup_interval}s, so spinning up a new process"
+          trigger_spinup = true
```

Comment:
I know the project convention is to have pretty minimal testing, but I notice that we at least have a `test/unit/einhorn/command.rb` file — do you think we could try to put a basic test in?

This will be somewhat difficult, so maybe we could do something relatively simple like moving all the logic below into a method called `trigger_spinup?` and then changing line 605 to:

``` ruby
if trigger_spinup?
  ...
```

`#trigger_spinup` would make a decision based off whether `max_additional_while_upgrading` is on and the current worker pool state. We could then at least test that method in isolation in the test suite with various inputs on it.

Reply:
I'm a bit skeptical that really will catch anything, but it's easy to do so

Reply:
It might not, but I think they're a good idea anyway because:

1. This is Ruby, and we have no idea whether code actually works until the moment you run it. This at least gives that path some basic exercise.
2. Tests (especially in RSpec-style) are a form of documentation, and they can help naive readers understand how pieces of code are expected to behave.

Anyway, thanks for adding!